src/__tests__/optimised-vary-large-performance-data.json

```diff
@@ -0,0 +1,1277 @@
+{
+    "navigation": [
+        {
+            "name": "http://127.0.0.1:8000/apm/waterfall",
+            "entryType": "navigation",
+            "duration": 938.3,
+            "initiatorType": "navigation",
+            "fetchStart": 6.5,
+            "domainLookupStart": 6.5,
+            "domainLookupEnd": 6.5,
+            "connectStart": 6.5,
+            "connectEnd": 6.5,
+            "requestStart": 18.8,
+            "responseStart": 155.7,
+            "responseEnd": 157,
+            "transferSize": 33689,
+            "encodedBodySize": 33389,
+            "decodedBodySize": 33389,
+            "unloadEventStart": 158.6,
+            "unloadEventEnd": 164.8,
+            "domInteractive": 346.5,
+            "domContentLoadedEventStart": 346.5,
+            "domContentLoadedEventEnd": 348.6,
+            "domComplete": 938,
+            "loadEventStart": 938,
+            "loadEventEnd": 938.3,
+            "type": "reload"
+        }
+    ],
+    "paint": [
+        {
+            "name": "first-paint",
+            "entryType": "paint",
+            "startTime": 310.6
+        },
+        {
+            "name": "first-contentful-paint",
+            "entryType": "paint",
+            "startTime": 310.6
+        }
+    ],
+    "resource": [
+        {
+            "name": "http://localhost:8234/static/index.css?t=1641815759815",
+            "startTime": 189.6,
+            "duration": 84,
+            "initiatorType": "link",
+            "fetchStart": 189.6,
+            "responseEnd": 273.6
+        },
+        {
+            "name": "http://localhost:8234/static/index.js?t=1641815759815",
+            "startTime": 256.3,
+            "duration": 21.2,
+            "initiatorType": "other",
+            "fetchStart": 256.3,
+            "responseEnd": 277.5
+        },
+        {
+            "name": "http://localhost:8234/static/chunk-AY3NAZV7.js",
+            "startTime": 256.7,
+            "duration": 33.6,
+            "initiatorType": "other",
+            "fetchStart": 256.7,
+            "responseEnd": 290.3
+        },
+        {
+            "name": "http://localhost:8234/static/chunk-POEQBYGC.js",
+            "startTime": 256.9,
+            "duration": 38,
+            "initiatorType": "other",
+            "fetchStart": 256.9,
+            "responseEnd": 294.9
+        },
```

Comment:
This is still not *that* optimised with ~50% of the body of each object being the same. What about something like:

```ts
function compressArrayOfObjects(objects: Record<string, any>[]) {
    if (objects.length === 0) { 
        return 
    }
    const keys = Object.keys(objects[0])
    return [
        keys,
        objects.map((obj) => keys.map(key => obj[key]))
    ]
}
```

... and a corresponding decompress function?

Reply:
That's really cool and brings the test case down to 20KB from 105KB 🙌